v03_pipeline/lib/misc/pedigree.py

```diff
@@ -77,17 +77,24 @@ def parse_direct_lineage(rows: list[hl.Struct]) -> dict[str, Sample]:
         for row in rows:
             # Maternal GrandParents
             maternal_s = samples[row.s].mother
-            if maternal_s and samples[maternal_s].mother:
-                samples[row.s].maternal_grandmother = samples[maternal_s].mother
-            if maternal_s and samples[maternal_s].father:
-                samples[row.s].maternal_grandfather = samples[maternal_s].father
+            if maternal_s and maternal_s not in samples:
+                # sample may be removed from the pedigree :/
```

Comment:
what does this mean?

Reply:
I can be more descriptive with the comment.  

It's basically just the pedigree getting subsetted from:
```
Individual_ID Paternal_ID Maternal_ID
sample_1       sample_2     sample_3
sample_2
sample_3
```

to

```
Individual_ID Paternal_ID Maternal_ID
sample_1       sample_2     sample_3
sample_3
```

so the parent is no longer present as a loadable sample.  It's also pretty straightforward to do this in airflow, just less testable.